identity/did.go

```diff
@@ -29,21 +25,83 @@ const (
 	// BootstrappedDIDService stores the id of the service
 	BootstrappedDIDService string = "BootstrappedDIDService"
 
-	// CentIDLength is the length in bytes of the DID
-	CentIDLength = 6
+	// KeyTypeECDSA has the value one in the ERC725 identity contract
+	KeyTypeECDSA = 1
 
-	// KeyPurposeP2P represents a key used for p2p txns
-	KeyPurposeP2P = 1
+	keyPurposeMgmt         = "MANAGEMENT"
+	keyPurposeAction       = "ACTION"
+	keyPurposeP2PDiscovery = "P2P_DISCOVERY"
+	keyPurposeSigning      = "SIGNING"
+)
 
-	// KeyPurposeSigning represents a key used for signing
-	KeyPurposeSigning = 2
+var (
+	// KeyPurposeManagement purpose stores the management key to interact with the ERC725 identity contract
+	KeyPurposeManagement Purpose
+	// KeyPurposeAction purpose stores the action key to interact with the ERC725 identity contract
+	KeyPurposeAction Purpose
+	// KeyPurposeP2PDiscovery purpose stores the action key to interact with the ERC725 identity contract
+	KeyPurposeP2PDiscovery Purpose
+	// KeyPurposeSigning purpose stores the action key to interact with the ERC725 identity contract
+	KeyPurposeSigning Purpose
+)
 
-	// KeyPurposeEthMsgAuth represents a key used for ethereum txns
-	KeyPurposeEthMsgAuth = 3
+func init() {
+	KeyPurposeManagement = getKeyPurposeManagement()
+	KeyPurposeAction = getKeyPurposeAction()
+	KeyPurposeP2PDiscovery = getKeyPurposeP2PDiscovery()
+	KeyPurposeSigning = getKeyPurposeSigning()
+}
 
-	// KeyTypeECDSA has the value one in the ERC725 identity contract
-	KeyTypeECDSA = 1
-)
+// getKeyPurposeManagement is calculated out of Hex(leftPadding(1,32))
+func getKeyPurposeManagement() Purpose {
+	enc := "0000000000000000000000000000000000000000000000000000000000000001"
+	v, _ := new(big.Int).SetString(enc, 16)
+	return Purpose{Name: keyPurposeMgmt, HexValue: enc, Value: v}
+}
+
+// getKeyPurposeAction is calculated out of Hex(leftPadding(2,32))
+func getKeyPurposeAction() Purpose {
+	enc := "0000000000000000000000000000000000000000000000000000000000000002"
+	v, _ := new(big.Int).SetString(enc, 16)
+	return Purpose{Name: keyPurposeAction, HexValue: enc, Value: v}
+}
+
+// getKeyPurposeP2PDiscovery is calculated out of Hex(sha256("CENTRIFUGE@P2P_DISCOVERY"))
+func getKeyPurposeP2PDiscovery() Purpose {
+	hashed := "88dbd1f0b244e515ab5aee93b5dee6a2d8e326576a583822635a27e52e5b591e"
+	v, _ := new(big.Int).SetString(hashed, 16)
+	return Purpose{Name: keyPurposeP2PDiscovery, HexValue: hashed, Value: v}
+}
+
+// getKeyPurposeSigning is calculated out of Hex(sha256("CENTRIFUGE@SIGNING"))
+func getKeyPurposeSigning() Purpose {
+	hashed := "774a43710604e3ce8db630136980a6ba5a65b5e6686ee51009ed5f3fded6ea7e"
+	v, _ := new(big.Int).SetString(hashed, 16)
+	return Purpose{Name: keyPurposeSigning, HexValue: hashed, Value: v}
+}
+
+// Purpose contains the different representation of purpose along the code
+type Purpose struct {
+	Name     string
+	HexValue string
+	Value    *big.Int
```

Comment:
I think we should not use the pointer here for Value. Since we are returning the Purpose as is, I can potentially change it even through a copy

Reply:
yikes! Good catch!